packages/rest/test/routes.test.ts

```diff
@@ -454,14 +454,59 @@ describe("rest: routes", () => {
       @test
       @autoRoute
       @sharedRoute
-      op get1(@path name: string): string;
+      op get1(@path foo: string): string;
 
       @test
       @autoRoute
-      op get2(@path name: string): string;
+      op get2(@path bar: string): string;
```

Comment:
Why did these names change?

Reply:
Because I got this error after merging in your shared route changes:

```
/test/main.tsp:10:7 - error @typespec/http/shared-inconsistency: All shared routes must agree on the value of the shared parameter.
```

This error feels reasonable but I can't exactly explain what the logic is.

Reply:
So there's a test for this. It means you couldn't have:
```
@route("foo/", {shared: true})
op test1()
@route("foo/", {shared: true})
op test2()
@route("foo/", {shared: false})
op test3()
```

So this should not be related to the name, though maybe because they are path parameters with autoroute it sees them as the same.

Reply:
I'll go double check and make sure I'm not introducing a new bug with `autoRoute` shared route support.

Reply:
Yep, `@autoRoute` causes both routes to end up as `/{name}` and since both are not marked as `@sharedRoute` the `shared-inconsistency` rule gets raised correctly.  Seems like everything is fine here, I'll just update the segment string on both `name` properties so that there's no route conflict.